File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        max_area=0;
        while(l<r):
            max_area = max(max_area,(min(height[r],height[l])*(r-l)))
            
            if(height[r]<height[l]):
                r-=1
            else:
                l+=1
        return max_area
            
        

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    long long maxArea(vector<int>& height) {
        long long int l=0;
        long long int r=height.size()-1;
        long long int max_area=0;
        while(l<r)
        {   
            max_area = max(max_area,(min(height[l],height[r]))*(r-l));
            
            if(height[l]<height[r])
                l++;
            else
                r--;
        }
        return max_area;
        
    }
};